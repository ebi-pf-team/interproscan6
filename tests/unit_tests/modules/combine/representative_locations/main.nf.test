nextflow_process {

    name "Test Process REPRESENTATIVE_LOCATIONS"
    script "modules/representative_locations/main.nf"
    process "REPRESENTATIVE_LOCATIONS"

    stage {
        symlink "tests/data/channels/matches2xrefs.json"
        symlink "tests/data/channels/matches_with_representative.json"
    }

    test("Should run without failures") {

        when {
            params {}
            process {
                """
                input[0] = [1, file("meta/tests/data/channels/matches2xrefs.json")]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
            def expected = "tests/data/channels/matches_with_representative.json"
            def observed = process.out.get(0).get(0).get(1)
            def cmd = "python3 tests/unit_tests/modules/combine/representative_locations/test_representatives.py --expected ${expected} --observed ${observed} --summary"
            def process = cmd.execute()
            process.waitFor()
            def stdout = process.text.trim().readLines()
            assert stdout[1] == "Matches only in expected : 0"
            assert stdout[2] == "Matches only in observed : 0"
        }
    }

}
