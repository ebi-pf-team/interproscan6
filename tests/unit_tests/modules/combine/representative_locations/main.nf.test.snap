{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        1,
                        "matches_with_representative.json:md5,40d94c6bc74e52202b783469a881d9c6"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-05-27T15:54:30.954917"
    }
}